src/steinertree/steiner_tree.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    filter_non_term_leaves!(g, term_vert)
+
+Remove edges of `g` so that all non-isolated leaves of `g` are in the set `term_vert`
+"""
+function filter_non_term_leaves!(
+    g::AbstractGraph{T},
+    term_vert::Vector{<:Integer}
+    ) where T<:Integer
+
+    is_term = falses(nv(g))
+    is_term[term_vert] .= true
+    leaves = [v for v in vertices(g) if degree(g, v) == 1 && !is_term[v]]
+
+    while !isempty(leaves)
+        leaf = pop!(leaves)
+        for v in neighbors(g, leaf)
+            rem_edge!(g, v, leaf)
+            if degree(g, v) == 1 && !is_term[v]
+                push!(leaves, v)
+            end
+        end
+    end
+
+    return g
+end
+
+"""
+	steiner_tree(g, term_vert, distmx=weights(g))
+
+Return a minimum weight spanning graph of connected, undirected graph `g` with positive edge 
+weights represented by `distmx` using [Approximate Steiner Tree](https://en.wikipedia.org/wiki/Steiner_tree_problem#Approximating_the_Steiner_tree).
+`t = length(term_vert)`
+
+### Performance
+Runtime: O(t*(t*log(t)+|E|*log(|V| ))
+Memory: O(t*|V|)
+Approximation Factor: 2-2/t
```

Comment:
Maybe explain the approximation factor in more details, is it just a lower or upper bound or does it go in both directions? And does that mean that `result_your_algorithm <= factor * result_correct` or
`factor * result_your_algorithm <= result_correct`.

Reply:
I don't mind but the explanation will have to be repeated for every approximation algorithm.
Instead, would it be better to explain approximation factor in the explanation of the Performance subsection?